hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -5075,17 +5103,22 @@ public boolean hasPathCapability(final Path path, final String capability)
       return isMagicCommitEnabled();
 
     case SelectConstants.S3_SELECT_CAPABILITY:
-      // select is only supported if enabled
-      return SelectBinding.isSelectEnabled(getConf());
+      // select is only supported if enabled and client side encryption is
+      // disabled
+      if(!isCSEEnabled) {
```

Comment:
simpler jut to merge with the return, eg.

```
return !isCSEEnabled && electBinding.isSelectEnabled(getConf());
```

Also, will `requireSelectSupport` need the same check?